src/run.rs

```diff
@@ -1,21 +1,21 @@
 use std::path::PathBuf;
+use ramemu::stmt::Stmt;
 use ramemu::{program::Program, ram::Ram};
-
 use crate::errors::RamCliError;
-use crate::errors::RamCliErrorKind;
 use crate::io_manager::create_input_buf;
 use crate::io_manager::create_output_buf;
 use crate::display_error::display_runtime_error;
 
 
 pub fn run_source (
     source: &str,
+    statements: Vec<Stmt>,
     input: Option<PathBuf>,
     output: Option<PathBuf>,
 ) -> Result<(), RamCliError> {
     let input_buf = create_input_buf(input)?;
     let output_buf = create_output_buf(output)?;
-    let program = Program::from_source(source).expect("Shuld be already checked");
+    let program = Program::from(statements);
```

Comment:
Your architecture seems wrong. You should not have separate function for checking correctness and for running. Moreover, checking should be the part of running.